foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/DockerPipInstallMigration.java

```diff
@@ -0,0 +1,143 @@
+package com.boozallen.aissemble.upgrade.migration.v1_7_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import com.boozallen.aissemble.upgrade.util.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.boozallen.aissemble.upgrade.util.FileUtils.getRegExMatches;
+
+public class DockerPipInstallMigration extends AbstractAissembleMigration {
+
+    private static final Logger logger = LoggerFactory.getLogger(DockerPipInstallMigration.class);
+    // starts with a negative look-ahead to ensure we are not matching lines that already contain
+    // "RUN set -e &&"
+    private static final String forDoPattern = "^(?!.*RUN set -e &&)RUN .*for .+; do .+; done";
+    private static final String findExecPattern =
+            "(^RUN )(find )(.*)( -exec)(.*python.* install (?:-\\S*\\s)*)(.+)";
+    private static final String RUN = "RUN ";
+    private static final String RUN_WITH_SET_E = "RUN set -e && ";
+    private static final String FILES_FIND = "files=$(find ";
+    private static final String FOR_FILE_IN_FILES_DO = "); for file in $files; do";
+    private static final String FILE_OR_EXIT_1_DONE = "$file || exit 1; done;";
+    private static final String upgradableInstructionsPattern =
+            String.join("|", findExecPattern, forDoPattern);
+
+    /**
+     * @param dockerFile the file to check
+     * @return boolean denoting if the dockerfile has at least one instruction to be migrated
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File dockerFile) {
+        boolean shouldExecute = false;
+
+        if (dockerFile != null && dockerFile.exists()) {
+            try {
+                ArrayList<String> upgradableDockerInstructions = getRegExMatches(
```

Comment:
A: Feels like overkill to find all of the matches just to figure out if at least one match exists.  Could just compile the pattern inline to a matcher and set `shouldExecute = matcher.find()`.  As a side note, we should probably not use a static method import here as it makes the code a bit harder to follow.  We tend to do it in tests because of the sheer number of times you have to call some static method from the Assert library, but here it's much less justified.

Reply:
That makes sense. Updating now